mediaire_toolbox/data_cleaner.py

```diff
@@ -18,35 +19,93 @@
 
 class DataCleaner:
 
-    def __init__(self, folder, max_folder_size, max_data_seconds):
+    def __init__(self, folder, max_folder_size, max_data_seconds,
+                 whitelist=None, blacklist=None):
+        """
+        Parameters
+        ----------
+        whitelist: list
+            Whitelist for files. Unix like filename pattern.
+        blacklist: list
+            Blacklist for files. Unix like filename pattern.
+        """
         self.base_folder = folder
         self.max_folder_size = max_folder_size
         self.max_data_seconds = max_data_seconds
+        self.whitelist = whitelist
+        self.blacklist = blacklist
+        if whitelist and blacklist:
+            raise ValueError("Both black list and white list in instance")
         default_logger.info("""Instantiated a DataCleaner on folder {%s} with max 
                     foldersize={%s} and max data seconds={%s}""" %
                     (folder, max_folder_size, max_data_seconds))
 
-    def current_size(self):
+    @staticmethod
+    def current_size(folder):
         """disk usage in bytes"""
-        return int(subprocess.check_output(['du', '-s', self.base_folder]) \
+        return int(subprocess.check_output(['du', '-s', folder]) \
             .split()[0] \
             .decode('utf-8'))
 
-    def creation_time(self, folder):
+    @staticmethod
+    def creation_time(folder):
         return int(os.path.getmtime(folder))
 
     @staticmethod
     def list_sub_folders(folder):
         return [os.path.join(folder, sub_folder)
                 for sub_folder in os.listdir(folder)]
 
+    def clean_folder(self, folder, dry_run=False):
+        """Cleans the folder given the filterlist.
```

Comment:
I'd find clearer something like "Cleans the contents of the parameter folder. If the instance was configured with a whitelist or a blacklist, a selective delete will be performed accordingly."

Reply:
Also if there is a return type in general this should be documented too